test/common/upstream/od_cds_api_impl_test.cc

```diff
@@ -130,10 +130,24 @@ TEST_F(OdCdsApiImplTest, OnDemandUpdateIsRequestedAfterInitialFetch) {
   const auto decoded_resources = TestUtility::decodeResources({cluster});
   odcds_callbacks_->onConfigUpdate(decoded_resources.refvec_, {}, "0");
   EXPECT_CALL(*cm_.subscription_factory_.subscription_,
-              requestOnDemandUpdate(UnorderedElementsAre("another_cluster")));
+              updateResourceInterest(UnorderedElementsAre("fake_cluster", "another_cluster")));
   odcds_->updateOnDemand("another_cluster");
 }
 
+// Check that we invoke requestOnDemandUpdate when no new cluster name.
```

Comment:
Can you also add an integration test, following the bug described in #24726?

Reply:
I had thought about it, it's hard to add such an integration test, since it depends on the subscription implementation.